### ISSUE TITLE
Bug: GeoX and GeoY points are transformed incorrectly

### ISSUE THREAD

**Issue by DilwoarH** [digital-land/submit#605]
## What

Ref: https://github.com/digital-land/submit/blob/main/src/models/responseDetails.js#L201-L204

- Remove hardcoded transformation of GeoX / GeoY to WKT
- Replace with transformation point in the `transformed_row` in the [Response details API](http://production-pub-async-api-lb-636110663.eu-west-2.elb.amazonaws.com/requests/9zSZptkGhvx2GoSJpoNHRK/response-details?offset=0&limit=1)

## Why

Some LPAs use BNG format and this is not compatible with WKT. This means when we convert this to show on the map, the map doesn't display the coordinates.

**Comment by DilwoarH** [comment]
Did initial investigation and the `transformed_row` property is not defined - I will push this back to the backlog so it can be prioritised.

cc @CharliePatterson @MJFaucherFolie @GeorgeGoodall-GovUk

**Comment by DilwoarH** [comment]
This was my code change:

```diff
 #makeGeometryGetter (item) {
    let getGeometryValue
    if ('point' in item.converted_row) {
      getGeometryValue = row => row.converted_row.point
    } else if ('geometry' in item.converted_row) {
      getGeometryValue = row => row.converted_row.geometry
    } else if ('GeoX' in item.converted_row) {
      logger.debug('converted_row', { type: types.App, transformedRow: item.transformed_row })
      getGeometryValue = row => {
-      const { GeoX, GeoY } = row.converted_row
-      if (GeoX === '' || GeoY === '') return ''
-      return `POINT (${GeoX} ${GeoY})`
+      const point = row?.transformed_row?.find(column => column.field === 'point')
+      return point ? point.value : ''
      }
    } else {
      // unexpected, but let's take a note and proceed without throwing
      logger.warn('geometry data not found in response details', { requestId: this.id, type: types.App })
    }

    return getGeometryValue
  }
````

**Comment by rosado** [comment]
The commit message from the original change has some clues: https://github.com/digital-land/submit/commit/9739e347ac7489ead2a0a5c787536d8960a06730

> We don't make use of the new 'transformed_row' entry in the body of
'/requests/:result-id/response-details' response (of async api),
because it's present only when 'dataset=brownfield-land' is passed
as param. The 'POINT' value is extracted manually from the GeoX, GeoY
columns (when present).

**Comment by rosado** [comment]
Some more details: seems like the sometimes the 'transformed_row' is there but has no geometry info that we expect.

For example using the test data linked below and submitting it to the check tool will give different results when you select 
- 'brownfield-lands' - you get the `point` in `transformed_row`
- 'article-4-direction' - no `point` in `transformed_row`

(I haven't checked all other datasets).

@ssadhu-sl confirmed that's how it works, what's left is to make the async-api work for datasets other that 'brownfield-lands'.

[Test data with GeoX+GeoY columns](https://gist.githubusercontent.com/rosado/3994e2420857d5c6b0e3bd8dc1488aa1/raw/bf1b52316f66a480a6d116c993ffd57d119094f9/geox-geoy-example.csv)

**Comment by DilwoarH** [comment]
@rosado is this still blocked?

**Comment by rosado** [comment]
@DilwoarH yes, infrastructure would need to make the endpoint include the `transformed_row` column for datasets other than 'brownfield-lands'

**Comment by ssadhu-sl** [comment]
Hi @rosado - _transformed_row_ exists in all the datasets. It's not specific to only brownfield-land. 
For datasets other than brownfield-land,  if geoX, geoY is sent instead, we should convert them to point. Can you send me an example of non-bfl dataset where geoX/geoY is sent.

**Comment by rosado** [comment]
@ssadhu-sl Try the data as described in https://github.com/digital-land/submit/issues/605#issuecomment-2488195980

I've just tried it again and submitting that file as 'article-4-direction' **does not** give us the `point` field in `transformed_row`

**Comment by ssadhu-sl** [comment]
@rosado article-4-direction does not have geoX, geoY fields mentioned in the specification ([check here](https://github.com/digital-land/specification/blob/main/content/dataset/article-4-direction.md))
Hence, it does not consider them when converting to a point. However, we do consider them when adding to the platform sometimes but for that we use pipeline/concat files but that we use only when actually adding to the platform. So a decision needs to be made on how we handle this case in check service.

**Comment by neilfwar** [comment]
fixed as we now use transformed rows instead of converted
